Fix bug 1691805: Do not make requests to MT services if locale not supported

mathjazz opened the pull request:
We're getting lots of 400 errors:
https://bugzilla.mozilla.org/show_bug.cgi?id=1691805

mathjazz commented:
@Pike r pretty please?

Pike commented on pontoon/machinery/views.py:
```diff
@@ -134,7 +134,10 @@ def microsoft_translator(request):
+    if (
+        not locale_code
+        or not Locale.objects.filter(ms_translator_code=locale_code).exists()
+    ):
```
Now that we only allow these requests for logged-in users, this query lost its purpose. I think we should just remove them.

Pike commented on pontoon/machinery/views.py:
```diff
@@ -240,6 +247,8 @@ def systran_translate(request):
             status=404,
         )
 
+    locale = Locale.objects.get(systran_translate_code=locale_code)
```
This could be ambiguous?

Pike reviewed (changes requested):
I wish this stuff worked differently, but that's also out of scope, I guess. If we'd eventually chain the MT engines server-side, we'd send the original locale code, and not the custom one. And then bridge that gap on the server side.

I think the frontend changes are OK, but the python parts bother me.

mathjazz commented:
I've addressed comments in https://github.com/mozilla/pontoon/pull/1862/commits/f654924aa8c08785e18ec31c7dc17eb5ddc3fa33 and https://github.com/mozilla/pontoon/pull/1862/commits/cffb0ea6368a5e070fd01d813c4bd481e05c1147.

In between I've made an additional refactoring.

Pike reviewed (approved):
Thanks, this makes more sense now :-)
